### PR TITLE
UI module

### PR DESCRIPTION
**Pull request by cquiroz**
This is a minimal (truly just a println) "UI" module. It basically sets the infrastructure so we can track the size as we add features. The PR contains:

* A new module compiled with scala.js that just does a println
* A update to the weigh script to track this new module's UI js
* Copied from ocs3 a task to upload the js to amazon s3 once merged. This will let us have a history of changes. Maybe useful in the future

This is just infrastructure. The idea is to start adding references to core classes from the "UI" and see the effects on the size as we go along

Feel free to discuss if this approach makes sense to you

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,13 +21,13 @@ install:
   - nvm install stable
   - nvm use stable
   - npm install jsdom
-
+  - pip install --user awscli
 script:
-  - sbt headerCheck test:headerCheck scalastyle sql/flywayMigrate compile test coreJS/test coreJS/fastOptJS
+  - sbt headerCheck test:headerCheck scalastyle sql/flywayMigrate compile test ui/fastOptJS ui/fullOptJS
```
Comment (swalker2m) [inline .travis.yml]: We won't run the test cases in JS now?

File: build.sbt [hunk 2]
```diff
@@ -186,7 +196,7 @@ lazy val flywaySettings = Seq(
 lazy val gem = project
   .in(file("."))
   .settings(commonSettings)
-  .aggregate(coreJVM, db, json, ocs2, ephemeris, service, telnetd, ctl, web, sql, main)
+  .aggregate(coreJVM, coreJS, db, json, ocs2, ephemeris, service, telnetd, ctl, web, sql, main, ui)
```
Comment (swalker2m) [inline build.sbt]: Size aside, how much does this add to compile time?

File: build/upload.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+# upload merged
+if [ -z "${AWS_ACCESS_KEY_ID}" ]; then
```
Comment (swalker2m) [inline build/upload.sh]: Is this an environment variable?

Comment (cquiroz) [reply build/upload.sh]: yes, it's only available when you merge though to keep it secret

File: .travis.yml (hunk 1 above)
Comment (cquiroz) [reply .travis.yml]: they should run implicitly now with the `aggregate` change

File: build.sbt (hunk 2 above)
Comment (cquiroz) [reply build.sbt]: Travis shows maybe 1m more though it varies, in some travis runs is even faster but that's likely due to the way travis work. This number will increase though as we evolve the ui especially `fullOptJS`

**Comment by tpolecat** [comment]
Can you explain what we're doing with s3 here?

**Comment by tpolecat** [comment]
This seems fine to me but I need convincing re: uploading to s3 if there's no reason to do so.

**Comment by cquiroz** [comment]
it seems we could get the same result without uploading the files but using a script to track the sizes instead, so I'm removing the S3 upload but keep the ui module

**Pull request merged**